examples/shouldi/shouldi/dependency_check.py

```diff
@@ -0,0 +1,68 @@
+import json
+import os
+import tempfile
+from pathlib import Path
+import asyncio
+from typing import Dict, Any
+
+from dffml.df.base import op
+from dffml.df.types import Definition
+
+package_src_dir = Definition(name="package_src_dir", primitive="str")
+dependency_check_output = Definition(
+    name="dependency_check_output", primitive="Dict[str, Any]"
+)
+tempdir = tempfile.TemporaryDirectory()
+
+
+@op(
+    inputs={"pkg": package_src_dir},
+    outputs={"report": dependency_check_output},
+)
+async def run_dependency_check(pkg: str) -> Dict[str, Any]:
+    """
+    CLI usage: dffml service dev run -log debug shouldi.dependency_check:run_dependency_check -pkg .
+    """
```

Comment:
tempdir should be a local variable. We want to use the `with` syntax to have it be removed after we're done running dependency check. See https://docs.python.org/3/library/tempfile.html?highlight=temporarydirectory#examples for how that works